src/main/js/providers/projects-provider.js

```diff
@@ -29,13 +30,69 @@ export default class ProjectsProvider extends React.Component {
         }));
     };
 
+    userAdded = (projectId, user, role) => {
+        this.setState(prevState => ({
+            projects: prevState.projects.map(project => {
+                if (project.id === projectId) {
+                    if (role === 'ROLE_LOCAL_COORDINATOR') {
+                        project.localCoordinator = user;
+                    } else if (role === 'ROLE_PUBLISHER') {
+                        project.publishers = project.publishers.concat([user]);
+                        project.publishers.sort((a, b) => {
+                            if (a.lastName < b.lastName) {
+                                return -1;
+                            } else if (a.lastName > b.lastName) {
+                                return 1;
+                            } else {
+                                return a.firstName < b.firstName ? -1 : 1;
+                            }
+                        });
+                    }
+                }
+                return project;
+            })
+        }));
+    };
+
+    userUpdated  = (user) => {
+        this.setState(prevState => ({
+            projects: prevState.projects.map(project => {
+                if (project.localCoordinator && project.localCoordinator.id === user.id) {
+                    project.localCoordinator = user;
+                }
+                if(project.publishers){
+                    project.publishers = project.publishers.map(tmpUser => tmpUser.id === user.id ? user : tmpUser);
+                }
+                return project;
+            })
+        }));
+    }
+
+    userRemoved = (userId) => {
+        this.setState(prevState => ({
+            //TODO: rerender of UserComponenet isn't triggered
```

Comment:
Das TODO kann weg, oder?

Reply:
Wenn man den Lokalen Koordinator löscht funktioniert das noch nicht

Reply:
Habs mir nochmal angesehen. Das Problem ist, dass in UserComponent der user im State verwaltet wird. D.h. Änderungen aus den Props kommen nur beim **initialen** Rendern an.

Einen nicht sehr schönen Fix hab ich hier commitet: 74a8a8b
(Sollte imho aber vorerst reichen)